scripts/data/generate_data.py

```diff
@@ -347,17 +384,19 @@ def generate_data(mode: str, initial_height: int, num_blocks: int, fast: bool):
     utreexo_data = {}
 
```

Comment:
If you try to regenerate tests you will likely get different (incorrect) data, because of https://github.com/keep-starknet-strange/raito/pull/239/files#diff-3f1cf8fcbdbcbbf57d51bf5e96d680aa09c51b831d78dbce1f45c8b4a1141cd7R437

I suggest you do the following:
- Instead of `chain_state` introduce `prev_chain_state` and `next_chain_state`

```python
initial_chain_state = (
        fetch_chain_state_fast(initial_height)
        if fast
        else fetch_chain_state(initial_height)
)
prev_chain_state = None
chain_state = initial_chain_state

# handle block ...

prev_chain_state = chain_state
chain_state = apply_chain_state(chain_state, block)

# finally

 result = {
        "chain_state": format_chain_state(initial_chain_state),
        "blocks": list(map(block_formatter, blocks)),
        "expected": format_chain_state(chain_state),
    }

if mode == "utreexo":
        utreexo_data = UtreexoData()
        result["utreexo"] = utreexo_data.apply_blocks(blocks)
        result["blocks"] = [result["blocks"][-1]]
        if num_blocks > 1:
            result["chain_state"] = format_chain_state(prev_chain_state)

```

Reply:
To verify I suggest to regenerate tests (at least one from each group light,full,utreexo).
How to do that? Just delete specific test files and run `scarb run regenerate_tests` in client folder

Reply:
Fix ser